### PR TITLE
Include linting in tox

### LINKED ISSUE #88: Add lint/setupcheck tests to Travis run
The `test.sh` script does linting and a few other things, however it's not run on Travis (and is failing for me locally under MSYS2), so hard to figure out if updating dependencies has broken anything (in #87).

Ideally those steps would be defined in tox.ini and run via tox both locally and on travis.

### PR DESCRIPTION
**Pull request by jhford**
Basically, I'd like to move as much as I can into tox.ini.  This should resolve #88 and also do a bit of general cleanup that's overdue.

### CODE REVIEW

**Comment by jhford** [comment]
@edmorley fyi

**Comment by djmitche** [comment]
Is this ready for review?  I'm not sure what is being ignored..

**Comment by jhford** [comment]
Please ignore README.md and genDocs.py, i will fix those.  it's probably related to how I'm handling utf-8 encoding and decoding.  Otherwise, the rest of the code is ready for review.

File: taskcluster/async/scheduler.py [hunk 1]
```diff
@@ -1,293 +0,0 @@
-# coding=utf-8
```
Comment (djmitche) [inline taskcluster/async/scheduler.py]: And, it looks like, remove scheduler support?  That should probably be broken out into a separate commit so it can be seen in the git log.
